src/basilisp/core.lpy

```diff
@@ -3258,6 +3258,21 @@
              (transient {})
              maps))))
 
+(defn merge-with
+  "Merge ``maps`` together from left to right, while reducing values
+  under the same key by successively applying function ``f`` on the
+  preceding and subsequent key values, i.e.
+
+  (``f`` (``f`` (``f`` first-value second-value) third-value) ...)."
```

Comment:
```suggestion
  "Merge maps together from left to right as by :lpy:fn:`conj`. If a duplicate key
  appears in a map, the resulting value of that key in the merged result will be the
  result of calling ``(f current-val next-val)``."
```

Sorry I found the proposed docstring a bit confusing. This follows more closely with the existing `merge` docstring.

Reply:
Updated.